### PR TITLE
Scale result csv

### PR DESCRIPTION
**Pull request by kornys**
### Type of change

- Enhancement / new feature

### Description

store results from scale tests into csv files for generating plots

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java [hunk 1]
```diff
@@ -372,6 +385,11 @@ public void testFailureRecovery() throws Exception {
 
         } finally {
             saveResultsFile("failure_recovery_results.json", manager.getDowntimeResult());
+            Map<String, Object> data = new HashMap<>();
+            data.put("addresses", addresses.size());
+            data.put("connections", manager.getConnections());
+            data.put("normalTimeToCreateAddress", manager.getDowntimeResult().getNormalTimeToCreateAddress());
+            savePlotCSV("failure_recovery.csv", data);
```
Comment (famartinrh) [inline systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java]: all the information related to specific downtime per pod removed is missing, however I understand that could be hard to represent in a csv. I would represent this with one row per pod removed.

File: systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java [hunk 2]
```diff
@@ -601,5 +619,17 @@ private void saveResultsFile(String filename, Object resultsObject) throws Excep
         Files.write(logsPath.resolve(filename), mapper.writeValueAsBytes(resultsObject));
     }
 
+    private void savePlotCSV(String fileName, Map<String, Object> data) throws IOException {
```
Comment (famartinrh) [inline systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java]: this only represents one row CSV, I think to get multiple lines you would need a List<Map<String, Object>> data

File: systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java [hunk 3]
```diff
@@ -257,6 +261,14 @@ void testMessagingPerformance() throws Exception {
             manager.stopMonitoring();
             manager.gatherPerformanceResults();
             saveResultsFile("messaging_performance_results.json", manager.getPerformanceResults());
+            Map<String, Object> data = new HashMap<>();
+            data.put("addresses", addresses.size());
+            data.put("connections", manager.getConnections());
+            data.put("anycastLinksPerConnection", anycastLinksPerConnection);
+            data.put("queueLinksPerConnection", queueLinksPerConnection);
+            data.put("anycastMedianThroughput", manager.getPerformanceResults().getAddresses().get(AddressType.ANYCAST.toString()).getGlobalSenders().getEstimateTotalThroughputMedian());
```
Comment (famartinrh) [inline systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java]: here you are only giving results for senders, is important to include receivers

File: systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java (hunk 2 above)
Comment (kornys) [reply systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java]: it must represent one line in csv, plot in jenkins shows continuous progress across builds, so every build is one line

File: systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java (hunk 3 above)
Comment (kornys) [reply systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java]: ok makes sense

File: systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java (hunk 1 above)
Comment (kornys) [reply systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java]: as I mention above, csv must be single line, so I cant add data like this, I only can list all pods and add them as columns into csv @famartinrh

Comment (famartinrh) [reply systemtests/src/test/java/io/enmasse/systemtest/scale/ScaleTest.java]: ok, let me think of it, and I could give a summarized result that fits into one line

**Pull request merged**